ground/src/main/java/com/google/android/ground/ui/datacollection/PhotoTaskFragment.kt

```diff
@@ -108,41 +115,13 @@ class PhotoTaskFragment : AbstractFragment(), TaskFragment<PhotoTaskViewModel> {
       .subscribe { photoResult -> viewModel.onPhotoResult(photoResult) }
   }
 
-  private fun onShowPhotoSelectorDialog() {
-    val addPhotoBottomSheetBinding: EditSubmissionBottomSheetBinding =
-      EditSubmissionBottomSheetBinding.inflate(layoutInflater)
-    val bottomSheetDialog = BottomSheetDialog(requireContext())
-    bottomSheetDialog.setContentView(addPhotoBottomSheetBinding.root)
-    bottomSheetDialog.setCancelable(true)
-    bottomSheetDialog.show()
-
-    val recyclerView: RecyclerView = addPhotoBottomSheetBinding.recyclerView
-    recyclerView.setHasFixedSize(true)
-    recyclerView.layoutManager = LinearLayoutManager(context)
-    recyclerView.adapter = AddPhotoDialogAdapter { type: Int ->
-      bottomSheetDialog.dismiss()
-      onSelectPhotoClick(type, viewModel.task.id)
-    }
-  }
-
-  private fun onSelectPhotoClick(type: Int, fieldId: String) {
-    when (type) {
-      PhotoStorageResource.PHOTO_SOURCE_CAMERA ->
-        // TODO: Launch intent is not invoked if the permission is not granted by default.
-        viewModel
-          .obtainCapturePhotoPermissions()
-          .andThen(Completable.fromAction { launchPhotoCapture(fieldId) })
-          .`as`(autoDisposable<Any>(viewLifecycleOwner))
-          .subscribe()
-      PhotoStorageResource.PHOTO_SOURCE_STORAGE ->
-        // TODO: Launch intent is not invoked if the permission is not granted by default.
-        viewModel
-          .obtainSelectPhotoPermissions()
-          .andThen(Completable.fromAction { launchPhotoSelector(fieldId) })
-          .`as`(autoDisposable<Any>(viewLifecycleOwner))
-          .subscribe()
-      else -> throw IllegalArgumentException("Unknown type: $type")
-    }
+  private fun onTakePhoto() {
+    // TODO: Launch intent is not invoked if the permission is not granted by default.
+    viewModel
+      .obtainCapturePhotoPermissions()
+      .andThen(Completable.fromAction { launchPhotoCapture(viewModel.task.id) })
+      .`as`(autoDisposable<Any>(viewLifecycleOwner))
+      .subscribe()
```

Comment:
I believe this will interrupted if the returned `Disposable` is GCed early.. Rather than managing the subscription, we can convert this to a coroutine and launch in application scope. Wdyt?